### PR TITLE
ref: use any as the key type WithValue

### PR DESCRIPTION
**Pull request by mdanialr**
## 📑 Description

Relate to https://github.com/goravel/framework/pull/628

@coderabbitai summary

## ✅ Checks

- [x] Added test cases for my code

### CODE REVIEW

**Review by devhaozi — DISMISSED** [review]

File: context.go [hunk 1]
```diff
@@ -39,8 +39,8 @@ func (c *Context) Response() http.ContextResponse {
 	return NewContextResponse(c.instance, &ResponseOrigin{Ctx: c.instance})
 }
 
-func (c *Context) WithValue(key string, value any) {
-	ctx := context.WithValue(c.instance.UserContext(), ctxKey(key), value)
+func (c *Context) WithValue(key any, value any) {
+	ctx := context.WithValue(c.instance.UserContext(), key, value)
```
Comment (hwbrzzl) [inline context.go:43]: Could you add some test cases with different keys? And Can the `Can we remove this?` be removed?

File: context_test.go [hunk 2]
```diff
@@ -14,3 +14,16 @@ func TestContext(t *testing.T) {
 	assert.Equal(t, httpCtx.Value("Hello").(string), "world")
 	assert.Equal(t, httpCtx.Value("Hi").(string), "Goravel")
 }
+
+func TestContextWithCustomKeyType(t *testing.T) {
+	type customKeyType struct{}
+	var customKey customKeyType
+	var customKeyTwo customKeyType
```
Comment (hwbrzzl) [inline context_test.go:21]: Please add additional other type keys: int, string, etc.

Comment (hwbrzzl) [reply context_test.go:21]: We may merge these two test cases: `TestContext`, `TestContextWithCustomKeyType`.

Comment (mdanialr) [reply context_test.go:21]: did you mean one function Test but with many sub-test @hwbrzzl ?

Comment (hwbrzzl) [reply context_test.go:21]: Yes, the test cases are very simple, two functions are unnecessary.

Comment (mdanialr) [reply context_test.go]: done in https://github.com/goravel/fiber/pull/102/commits/9989eb4ce2ee4875e23cb85080f647f2f038b05e

File: context.go (hunk 1 above)
Comment (mdanialr) [reply context.go:41]: done in https://github.com/goravel/fiber/pull/102/commits/9799af9b05707826c7cd304de6a893d56b704f0b

**Pull request closed without merging**